packages/migration/first_ecosys_contracts_data.go

```diff
@@ -130,15 +132,25 @@ VALUES ('2', 'DelApplication', 'contract DelApplication {
 			}
 			i = i + 1
 		}
+
+		$row = DBFind("languages").Columns("name,app_id").WhereId($Id).Row()
+		if !$row{
+			warning "Language not found"
+		}
+
+		if $ApplicationId == 0 {
+			$ApplicationId = Int($row["app_id"])
+		}
+		if $Name == "" {
+			$Name = $row["name"]
+		}
+
 		if (len > 0){
 			langarr = Sprintf("{"+"%%v"+"}", res)
 			$Trans = langarr
-			DBUpdate("languages", $Id, "res", $Trans)
-			//UpdateLang($Id, $Trans)
-		}
-		else {
-			DBUpdate("languages", $Id, "res", $Trans)
-			//UpdateLang($Id, $Trans)
+			EditLanguage($Id, $Name, $Trans, $ApplicationId)
+		} else {
+			EditLanguage($Id, $Name, $Trans, $ApplicationId)
```

Comment:
What the point having that two strings that looks the same?
maybe something like:
if (len > 0) {
   $Trans = Sprintf("{"+"%%v"
}
EditLanguage($Id, $Name, $Trans, $Application)